rust/common/cache/src/types.rs

```diff
@@ -0,0 +1,233 @@
+//! Cache configuration types and cache operation results
+//!
+//! This module contains the core types used throughout the cache system:
+//! - [`CacheConfig`]: Configuration for cache instances (prefix, TTL)
+//! - [`CacheSource`]: Enum indicating where a value came from (for observability)
+//! - [`CacheResult`]: Wrapper containing a value and its source
+
+use std::fmt;
+
+/// Configuration for cache instances
+#[derive(Debug, Clone)]
+pub struct CacheConfig {
+    /// Redis key prefix for this cache instance (e.g., "team_token:", "feature_flags:")
+    pub cache_prefix: String,
+
+    /// Optional TTL in seconds for cached values
+    /// If None, values are cached indefinitely
+    pub ttl_seconds: Option<u64>,
+}
+
+impl CacheConfig {
+    /// Create a new cache configuration
+    pub fn new(cache_prefix: impl Into<String>, ttl_seconds: Option<u64>) -> Self {
+        Self {
+            cache_prefix: cache_prefix.into(),
+            ttl_seconds,
+        }
+    }
+
+    /// Create a cache configuration with no TTL (permanent caching)
+    pub fn permanent(cache_prefix: impl Into<String>) -> Self {
+        Self::new(cache_prefix, None)
+    }
+
+    /// Create a cache configuration with a TTL
+    pub fn with_ttl(cache_prefix: impl Into<String>, ttl_seconds: u64) -> Self {
+        Self::new(cache_prefix, Some(ttl_seconds))
+    }
+}
+
+/// Indicates where a cached value came from and what operations were performed
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
```

Comment:
could add a docs test here if you wanted, e.g.
```
 /// Indicates where a cached value came from and what operations were performed
+///
+/// This type implements `Display` for use in logging and metrics:
+/// ```
+/// # use common_cache::CacheSource;
+/// let source = CacheSource::PositiveCache;
+/// println!("Cache result: {}", source); // "positive_cache"
+/// ```
``